Try using index.js if 'main' in package.json is a directory

kipz opened the pull request:
This is happening more and more in modules in the wild, and Node seems to deal with this totally fine.

malaporte commented on src/test/java/com/coveo/nashorn_modules/ModuleTest.java:
```diff
@@ -548,4 +548,12 @@ public void itIncludesFilenameInException() throws Throwable {
   }
+
+  @Test
+  public void itLoadsIndexJsIfMainDirSpecified() throws Throwable {
```
Nitpicking, but there are already a few tests that cover various use cases like this around line 188; it'd be cool to have this one in the same spot. Also, those tests use mocks instead of static resources to setup each use case. Maybe do the same here? It requires almost no code makes the test a bit easier to understand I think.

kipz commented:
Thanks for the quick turn around. I've moved the test and converted it to using mocks as requested.

I really appreciate you taking the time to consider our contributions!

kipz commented on src/main/java/com/coveo/nashorn_modules/Module.java:
```diff
@@ -133,7 +133,7 @@ public Object require(String module) throws ScriptException {
     } finally {
       // Finally, we remove the successful resolved module from the refCache
-      if(requestedFullPath != null){
+      if (requestedFullPath != null) {
```
This is a formatting error that snuck in on my last PR!

kipz commented:
Thanks for merging.

There are a few useful changes in master right now, so I was wondering if you were considering a release any time soon?

malaporte commented:
Well I tried to do one last weekend, but for some reason my GnuPG installation was all borked up. I just found the issue: it was missing an env variable to point it to my TTY, which it uses to ask me for my passphrase. Funny thing is that it never asked me for it before, on this very laptop, but hey.

Release should be available shortly.

kipz commented:
Much appreciated. I'm grabbing it now!
